e2e/framework/pages/details/details-page.ts

```diff
@@ -0,0 +1,40 @@
+import { Page } from "playwright";
+import { PageWithNavMenuAndMessages } from "../_common-page/common-page";
+import { waits } from "../../../libs/helpers/waits";
+
+export class DetailsPage extends PageWithNavMenuAndMessages {
+    private heroNameInputFieldSelector = '#hero-name';
+
+    constructor(page: Page) {
+        super(page);
+    }
+
+    async getHeroId(): Promise<string | undefined> {
+        let allTexts = await (await waits(this.page).waitVisibility('//div[child::span]')).textContent();
+        return allTexts?.split(":")[1].trim();
+    }
+
+    async getHeroName(): Promise<string> {
+        return await this.page.locator(this.heroNameInputFieldSelector).inputValue();
+    }
+
+    async fillNewHeroName(name: string): Promise<DetailsPage> {
+        await this.page.locator(this.heroNameInputFieldSelector).fill(name);
+        return this;
+    }
+
+    async clickSaveButton(): Promise<void> {
+        await this.page.locator('text=save').click();
+    }
+
+    async clickGoBackButton(): Promise<void> {
+        await this.page.locator('text=go back').click();
+    }
+
+    async fillNameAndClickSaveButton(name: string): Promise<void> {
+        await Promise.all([
+            this.fillNewHeroName(name),
+            this.clickSaveButton()
```

Comment:
`Promise.all` executes in parallel, this code might click save before filling name. Is this intended?
If not just do
```
await this.fill(...);
await this.clickSave(...);
```

Reply:
Fixed.